app/kuma-ui/pkg/resources/data/index.html

```diff
@@ -1,15 +1,18 @@
 <!DOCTYPE html>
 <html lang="en">
   <head>
+    <script id="kuma-config" type="application/json">
+      {{.}}
+    </script>
     <meta charset="utf-8" />
     <meta name="viewport" content="width=device-width,initial-scale=1" />
 
-    <link rel="icon" href="./favicon.png" />
+    <link rel="icon" href="/{{.BaseGuiPath}}/favicon.png" />
```

Comment:
Just noticed, I'm pretty sure this could be better if we make `BaseGuiPath` absolute, i.e. include the `/` at the beginning. 

It has to be absolute anyway seeing as we can access the GUI from any number of folders deep (for the browser deep-linking/bookmarking/refreshing purposes)

As it stands at the moment the BaseGuiPath used here is `/gui` (as we statically prepend the `/`) whereas in the JSON that the GUI/JS will read it is `gui` _without the `/`_, which is probably an important detail.

For example given `BaseGuiPath='gui'` landing at the GUI on `/mesh/default/data-planes` would mean that inside JS the baseGUIPath would resolve to `/mesh/default/gui` when it should be `/gui`. 

See here for an example of the non-absolute baseGuiPath:

https://github.com/kumahq/kuma/pull/5357/files#diff-ed0ed0fa97d6799cb868041619cb13ce4a66072ee40b8afb81675b5cadc39ca8R5

What do you think? Does the above make sense?

Reply:
This was done so on my explicit request. We can’t set Vite’s `base` parameter to `'{{.BaseGuiPath}}'` without it adding slashes around it.

So, long story short, `{{.BaseGuiPath}}` may not contain leading or trailing slashes.

Reply:
> For example given BaseGuiPath='gui' landing at the GUI on /mesh/default/data-planes would mean that inside JS the baseGUIPath would resolve to /mesh/default/gui when it should be /gui.

No? `BaseGuiPath` would always be "gui". Its value doesn’t change depending on request URL.

Reply:
I’ve added a comment in https://github.com/kumahq/kuma-gui/pull/468/files#diff-58e6f63d87181b1c6a8cb6e5f1691df04aa32854456efcd52ca71c8541375d26R9 to make this explicit.

Reply:
> So, long story short, {{.BaseGuiPath}} may not contain leading or trailing slashes.

So you're saying this is due to restrictions around vite's configuration?

> No? BaseGuiPath would always be "gui". Its value doesn’t change depending on request URL.

I suppose it depends what you are using it for. If you try to load an image at `gui/image.png` that's a completely different thing to loading it from `/gui/image.png`

Let me sit with this a little brb

Reply:
Would it be ok if the vite config was:

```
base: '/{{slice .BaseGuiPath 1}}/'
```

That means that the JSON blob would have an absolute path in it:

```
{
  BaseGuiPath: '/ui'
}
```

?

Reply:
> So you're saying this is due to restrictions around vite's configuration?

Yes, but not something that bothers us. It’s perfectly fine to substitute this variable with a path that doesn’t have a leading or trailing slash.

> I suppose it depends what you are using it for. If you try to load an image at gui/image.png that's a completely different thing to loading it from /gui/image.png

While that is a difference, we deliberately structure the paths such that these URLs are always relative to the web root. Currently, we refer to assets relative to the a location using `./assets/...`. In the future, we will always refer to `/$guiPath/assets/...` where `/$guiPath/` is relative to the origin/web root. So for the quasi default GUI URL off http://localhost:5681/gui you’d end up with `$guiPath` set to `gui` and asset reference like `/gui/assets/...`.

Reply:
That'd be great too as it would avoid this weird trim: https://github.com/kumahq/kuma/blob/6bc5b5207429637ef4000abe71bc735f558a908c/pkg/api-server/server.go#L182

Reply:
Personally, I’d really rather not have two different GUI paths.